website/static/js/fangorn.js

```diff
@@ -236,6 +250,105 @@ function _fangornToggleCheck(item) {
+                m('button.btn.btn-md.btn-default.m-r-sm', {onclick: tb.modal.dismiss.bind(tb)}, 'Cancel'),
+                m('button.btn.btn-md.btn-default', {onclick: cb.bind(tb, 'replace')},'Replace'),
+            ]));
+            return;
+        }
+    }
+    cb('replace');
+}
+
+function doItemOp(isMove, to, from, rename, conflict) {
+    var tb = this;
+    tb.modal.dismiss();
+    var ogParent = from.parentID;
+    if (to.id === ogParent && (!rename || rename === from.data.name)) return;
+
+    if (isMove) {
+        from.move(to.id);
+        from.data.status = 'move';
+    } else {
+        from = tb.createItem($.extend(true, {}, from.data), to.id);
+        from.data.status = 'copy';
+    }
+
+    tb.redraw();
+
+    $.ajax({
+        type: 'POST',
+        url: isMove ? waterbutler.moveUrl() : waterbutler.copyUrl(),
+        headers: {
+            'Content-Type': 'Application/json'
+        },
+        data: JSON.stringify({
+            'rename': rename,
+            'conflict': conflict,
+            'source': waterbutler.toJsonBlob(from),
+            'destination': waterbutler.toJsonBlob(to),
+        })
+    }).done(function(resp, _, xhr) {
+        if (xhr.status === 202) {
+            var mithrilContent = m('div', [
+                m('h3.break-word', (isMove ? 'Moving' : 'Copying') + ' "' + from.data.materialized + '" to "' + (to.data.materialized || '/') + '" is taking a big longer than expected'),
+                m('p', 'We\'ll send you an email when it has finished.')
+            ]);
+            var mithrilButtons = m('div', [
+                m('span.tb-modal-btn', { 'class' : 'text-default', onclick : function() { tb.modal.dismiss(); }}, 'OK')
+            ]);
+            tb.modal.update(mithrilContent, mithrilButtons);
+            return;
+        }
+        from.data = resp;
+        from.data.status = undefined;
+        from.notify.update('Successfully ' + (isMove ? 'moved.' : 'copied.'), 'success', null, 1000);
+
+        if (!isMove && xhr.status === 200) {
+            to.children.forEach(function(child) {
+                if (child.data.name === from.data.name && child.id !== from.id) {
+                    child.removeSelf();
+                }
+            });
+        }
+
+        inheritFromParent(from, from.parent());
+
+        if (from.data.kind === 'folder' && from.data.children) {
+            from.children = [];
+            var child;
+            from.data.children.forEach(function(item) {
+                child = tb.buildTree(item, from);
+                inheritFromParent(child, from);
+                from.add(child);
+            });
+            from.open = true;
+            from.load = true;
+        }
+
+        tb.redraw();
+    }).fail(function() {
+        if (isMove) {
+            from.move(ogParent);
+            from.data.status = undefined;
+        } else {
+            from.removeSelf();
+        }
+
+        $osf.growl('Error', (isMove ? 'Move' : 'Copy') + ' failed.');
```

Comment:
This error message could be more descriptive. At the very least, include the support email, e.g. "If the problem persists, please contact support@osf.io".